### PR TITLE
V9: SqlCeImageMapper only applied for SqlCe

### PR DESCRIPTION
**Pull request by bergmania**
Introduced `IProviderSpecificMapperFactory`, to allow some NPoco mappers to only be applied for specific providers, e.g. `SqlCe` and the `SqlCEImageMapper`.

Before this we saw issues with mapping `byte[]` when using `SqlCE`. Because the `SqlCeImageMapper `was not added to the mappers when using `UmbracoDatabase`.

I had to introduce a `IProviderSpecificMapperFactory `to ensure the `SqlCEImageMapper `was only applied when using `SqlCE`, because it failed otherwise.

### CODE REVIEW

**Review by Shazwazza — CHANGES_REQUESTED** [review]
I might be totally missing something but I don't see any place where the new method `ProviderSpecificMappers` that resolves the mappers is actually used?

File: src/Umbraco.Infrastructure/Persistence/UmbracoDatabaseFactory.cs [hunk 1]
```diff
@@ -280,6 +280,8 @@ private SqlContext Initialize()
             // add all registered mappers for NPoco
             _pocoMappers.AddRange(_npocoMappers);
 
+            _pocoMappers.AddRange(_npocoMappers);
```
[the remark this answers is not in the document]
Comment (bergmania) [reply src/Umbraco.Infrastructure/Persistence/UmbracoDatabaseFactory.cs:283]:
```suggestion
            _pocoMappers.AddRange(_dbProviderFactoryCreator.ProviderSpecificMappers(_providerName));
```

 Doh, this should have been like this suggestion, and thereby also answer your other question

File: src/Umbraco.Infrastructure/Persistence/IDbProviderFactoryCreator.cs [hunk 2]
```diff
@@ -10,5 +10,6 @@ public interface IDbProviderFactoryCreator
         ISqlSyntaxProvider GetSqlSyntaxProvider(string providerName);
         IBulkSqlInsertProvider CreateBulkSqlInsertProvider(string providerName);
         void CreateDatabase(string providerName);
+        NPocoMapperCollection ProviderSpecificMappers(string providerName);
```
[the remark this answers is not in the document]
Comment (bergmania) [reply src/Umbraco.Infrastructure/Persistence/IDbProviderFactoryCreator.cs:13]: See answer in the other comment

**Review by Shazwazza — APPROVED** [review]
Code looks good. I'm assuming you already tested this so happy for you to merge if so :)

**Pull request merged**